### PR TITLE
RMSW 42 - Implement torque control actuator

### PR DESCRIPTION
**Pull request by Arct1cPharaoh**
Still WIP
- Added torque control actuator
- Added torque control actuator test

I have many notes (ctrl+f NOTE:) and this is most definitely not done. Hoping for corrections on direction and functionality.
To run test:
`make tests/TorqueControlActuatorTest.out && ./tests/TorqueControlActuatorTest.out`

### CODE REVIEW

File: includes/TorqueControlActuator.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#ifndef RENSSELAERMOTORSPORT_TORQUE_CONTROL_ACTUATOR_H
+#define RENSSELAERMOTORSPORT_TORQUE_CONTROL_ACTUATOR_H
+
+#include "DigitalSensor.h"
+
+#define TORQUE_ERROR_MARGIN 0.05 // NOTE: IDK What this value should be realistically
```
[the remark this answers is not in the document]
Comment (AmarMaksumic) [reply includes/TorqueControlActuator.h:6]:
```suggestion
#define TORQUE_ERROR_MARGIN 0.05 // TODO: Determine correct value for the margin, or change to lookup table if needed.
```

**Review by AmarMaksumic — COMMENTED** [review]
Let me comeup with a flow diagram of how we should go about the torque control and leaving in space for the implementation of policing and controls (like traction). We might need to move some methods around for that to happen. Are you fine with me doing that?

**Comment by AmarMaksumic** [comment]
They are things I should have clarified before and would be better for the general system architecture. That being said, most of this will probably remain as is and gives us a good platform to start off of

**Review by Arct1cPharaoh — COMMENTED** [review]
Looked through all the changes and they look great. Nice work!

File: Makefile [hunk 2]
```diff
@@ -39,6 +39,9 @@ $(TEST_DIR)/%.out: $(TEST_DIR)/%.c $(OBJECTS)
 	@mkdir -p $(@D)
 	$(CC) $< $(OBJECTS) -o $@ $(CFLAGS)
 
+print-objects:
```
Comment (AmarMaksumic) [inline Makefile:42]: remove this

File: includes/Common.h [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#ifndef RENSSELAERMOTORSPORT_COMMON_H
+#define RENSSELAERMOTORSPORT_COMMON_H
+
+#define SUCCESS 0;
```
Comment (AmarMaksumic) [inline includes/Common.h:4]: add comment explaining what this file is

**Review by AmarMaksumic — APPROVED** [review]
Code does not work as is because it needs torque monitor in place, which is part of another PR.

**Pull request merged**